src/main/java/CommandParser.java

```diff
@@ -0,0 +1,84 @@
+package main.java;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandParser {
+    private String input;
+    private String type;
+
+    public CommandParser(String input) {
+        this.input = input;
+        int firstSpaceIndex = input.indexOf(" ");
+        if (firstSpaceIndex != -1) {
+            this.type = input.substring(0, firstSpaceIndex);
+        } else {
+            this.type = input;
+        }
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public String[] parseInput() throws ChatException {
+        switch (this.type) {
+            case "todo":
+                String regex1 = "^(\\w+)(\\s)(.*)";
```

Comment:
As mentioned [here](https://nus-cs2103-ay2324s2.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/) try not to use regex1, regex2, etc. Maybe name it todoRegex or just combine lines 27 and 28 into one.